trigger/payloads.go

```diff
@@ -451,7 +451,16 @@ func getFailedTestsFromTemplate(t task.Task) ([]task.TestResult, error) {
 	for i := range t.LocalTestResults {
 		if t.LocalTestResults[i].Status == evergreen.TestFailedStatus {
 			testResult := t.LocalTestResults[i]
-			testResult.URL = settings.Ui.Url + testResult.URL
+			logURL, err := url.Parse(testResult.URL)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to parse URL %s", testResult.URL)
+			}
+			if logURL.Host == "" {
```

Comment:
how does this work? 
Is the testresult url sometimes relative to Evergreen and sometimes its own URL?

Reply:
Correct. For logs that are stored in the evergreen db, it will be a relative URL, but ones in logkeeper will be an absolute URL

Reply:
Where do we set the relative URL for logs stored in our db? I see where we set the log id ([here](https://github.com/evergreen-ci/evergreen/blob/ed0a539162449b6580f58652b306ea9e109aea77/agent/command/results_utils.go#L75), [here](https://github.com/evergreen-ci/evergreen/blob/d4345d7205660e2ad059a05a8a5b1a7cd991f1b8/agent/command/results_native.go#L107), [here](https://github.com/evergreen-ci/evergreen/blob/d4345d7205660e2ad059a05a8a5b1a7cd991f1b8/agent/command/results_go_test2json.go#L114), [here](https://github.com/evergreen-ci/evergreen/blob/d4345d7205660e2ad059a05a8a5b1a7cd991f1b8/agent/command/results_xunit.go#L178)), but not where we set the URL.

Reply:
Ah good point, I looked but didn't see it. It looks like we form the URL from the log ID, so I did that here. I imagine this has never worked correctly. I actually get these notifications but the slack version doesn't have the log link :(